### ISSUE TITLE
contentFiles not working for newer project formats

### ISSUE THREAD

**Issue by PureWeen** [akavache/Akavache#385]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**

Currently when installing Akavache into new project format projects the AkavacheSqliteLinkerOverride files doesn't get added

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

Create a project using the PackageReference format and install Akavache (any version) you'll see the file doesn't get added

**What is the expected behavior?**

That AkavacheSqliteLinkerOverride  gets added and transformed

**What is the motivation / use case for changing the behavior?**
iOS links out SQLitePersistentBlobCache and then the static caches just use InMemoryCache

**Which versions of Akavache, and which platform / OS are affected by this issue? Did this work in previous versions of Akavache? Please also test with the latest stable and snapshot (https://www.myget.org/feed/Akavache/package/nuget/Akavache) versions.**

Anything that's new project format

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

I messed with this a little bit but couldn't figure out how to make it work...  

I read through this
https://docs.microsoft.com/en-us/nuget/schema/nuspec#including-content-files

And added the content files in the specified location
/cs/any/AkavacheSqliteLinkerOverride.cs.pp  
But it doesn't want to work

I read through this issue
https://github.com/NuGet/Home/issues/4803

and in there they reference
https://www.nuget.org/packages/ContentFilesExample/

Which also has a pp file included in the content that doesn't add into a destination project.  The best I was able to get working is if you don't use a pp file and just a cs file but even then I was only able to get it to add into netstandard projects and when I tried to add into a xamarin project the file didn't get copied into it.

It sounds like from the answers the solution is to add it via init.ps1 or a targets file but I might not be following the conversation right.
> The plan going forward for this is to allow users to move files from under the contentFiles folder into their project through a gesture in Visual Studio. Currently the include of these files is written out to the auto generated nuget props file in the obj folder which allows them to be removed or overridden.
>
>My suggestion for copying config XML files to the user project is to handle this in an init.ps1 script, or possibly in a .targets file.

**Comment by PureWeen** [comment]
I'm guessing this
https://github.com/dotnet/sdk/issues/1100

is a blocker for the pp files

**Comment by ghuntley** [comment]
Init.ps1 still works (this was the vector for that nuget typosquatting exploit) but it executes _everytime_ a solution is opened. It only works on Windows and when using Visual Studio for Windows. The other PowerShell options such as install.ps1 no longer work.

**Comment by PureWeen** [comment]
The best workaround i've found so far is to just include a cs file (no pp) into  contentfiles/any/any

Then it'll get added into the netstandard projects at least but not the head projects.

we could probably add a content readme into
contentfiles/any/xamarinios10
contentfiles/any/monoandroid7

which I'm hoping work since contentfiles adds based on TFM

That just warns about it getting linked out with a sample of the files to add

On a side note this isn't a breaking change for the 6.0 release ... 5.0 has the same issue when interacting with the new project formats

**Comment by ghuntley** [comment]
The prayer to the linker gods was from a time before options like https://developer.xamarin.com/guides/cross-platform/advanced/custom_linking/ existed. Wonder if we can package up a linker configuration definition into a nuget package?

**Comment by ghuntley** [comment]
Yes; we can release without resolving this as long as we update the README up around the top with some instructions/link to this GitHub issue.

**Comment by PureWeen** [comment]
I haven't been able to find a way to package anything up :-/  this one seems a bit special because it's the whole assembly being linked out since the only interaction the library has outward is via reflection..   Like even just calling an empty init method into the assembly fixes it

Maybe contentFiles supports a linker build action and will include a file.. I'll try it in a few just to see :-x

**Comment by PureWeen** [comment]
I'm wondering if it's worth just punting on this whole idea of inserting the linked file... Mainly because the issue isn't really with SqlPersistentBlobCache getting linked out the issue is with the DLL just not being included at all because of the static analysis of nothing calling into the DLL..    If you create an empty INIT method inside Akavache.SqlLite3 that literally does nothing and an outside library called into it it has the same affect as the Linked in file.    

Having to set the ApplicationName in order to use the static BlobCache properties is already init code that has to be added.    Plus all of this is only really relevant if you're just relying on the static Cache variables.  

I feel like just adding some Init call into Akavache.SqlLite3 is more explicit and just kills a few birds with one stone... Something like

```
namespace Akavache.SqlLite3
public void InitSqlLiteStoreForGlobalCaches(string ApplicationName)
{
BlobCache.ApplicationName = ApplicationName
//maybe even stick the batteries call here and add some properties on the method to include their own batteries
}
```

**Comment by glennawatson** [comment]
Thank you for reporting this. Closing as this is no longer applicable to Akavache V11.

V11 no longer ships `AkavacheSqliteLinkerOverride` content files or relies on NuGet contentFiles for linker preservation. The approach has changed completely:

- `[SQLite.Preserve(AllMembers = true)]` attribute is applied directly to `SqliteBlobCache`
- `IsAotCompatible = true` is set in project properties for net8.0+
- `[RequiresUnreferencedCode]` and `[RequiresDynamicCode]` attributes inform the trimmer about dynamic code requirements
- The old static `BlobCache` API that required linker preservation of `SQLitePersistentBlobCache` and `SQLiteEncryptedBlobCache` has been replaced with the builder pattern

V11 uses standard SDK-style projects with `PackageReference` exclusively — no contentFiles transformation needed.

If you encounter linker/trimmer issues with V11, please open a new issue.
